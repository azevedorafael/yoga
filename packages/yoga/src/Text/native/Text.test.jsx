import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '../..';
import { Text } from '.';

describe('<Text />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with Text', () => {
      const { container } = render(
        <ThemeProvider>
          <Text.H1>Live the mission</Text.H1>
          <Text.H2>Live the mission</Text.H2>
          <Text.H3>Live the mission</Text.H3>
          <Text.H4>Live the mission</Text.H4>
          <Text>Live the mission</Text>
          <Text.Small>Live the mission</Text.Small>
          <Text.Tiny>Live the mission</Text.Tiny>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with Text variant', () => {
      const { container } = render(
        <ThemeProvider>
          <Text.H1 variant="primary">Live the mission</Text.H1>
          <Text.H2 variant="secondary">Live the mission</Text.H2>
          <Text.H3 variant="tertiary">Live the mission</Text.H3>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with inverted Text', () => {
      const { container } = render(
        <ThemeProvider>
          <Text inverted>Live the mission</Text>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
