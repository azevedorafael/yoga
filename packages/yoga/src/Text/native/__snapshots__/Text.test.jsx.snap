// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Snapshots should match snapshot with Text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 32,
          "fontWeight": "700",
          "lineHeight": 40,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 24,
          "fontWeight": "700",
          "lineHeight": 32,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 20,
          "fontWeight": "700",
          "lineHeight": 24,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 20,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    inverted={false}
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#41414A",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 16,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
</View>
`;

exports[`<Text /> Snapshots should match snapshot with Text variant 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F46152",
          "fontSize": 32,
          "fontWeight": "700",
          "lineHeight": 40,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    variant="primary"
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#652CDB",
          "fontSize": 24,
          "fontWeight": "700",
          "lineHeight": 32,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    variant="secondary"
  >
    Live the mission
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#2B9486",
          "fontSize": 20,
          "fontWeight": "700",
          "lineHeight": 24,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    variant="tertiary"
  >
    Live the mission
  </Text>
</View>
`;

exports[`<Text /> Snapshots should match snapshot with inverted Text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    inverted={true}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    Live the mission
  </Text>
</View>
`;
